x-pack/legacy/plugins/apm/server/lib/apm_telemetry/collect_data_telemetry/tasks.ts

```diff
@@ -0,0 +1,481 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { flatten, merge, set, sum } from 'lodash';
+import { AGENT_NAMES, AgentName } from '../../../../common/agent_name';
+import { Transaction } from '../../../../typings/es_schemas/ui/Transaction';
+import {
+  PROCESSOR_EVENT,
+  SERVICE_NAME,
+  SERVICE_AGENT_NAME,
+  ERROR_GROUP_ID,
+  TRANSACTION_NAME,
+  PARENT_ID
+} from '../../../../common/elasticsearch_fieldnames';
+import { Span } from '../../../../typings/es_schemas/ui/Span';
+import { APMError } from '../../../../typings/es_schemas/ui/APMError';
+import { TelemetryTask } from './';
+import { APMTelemetry } from '../types';
+
+const TIME_RANGES = ['1d', '1M', '6M', 'all'] as const;
+type TimeRange = typeof TIME_RANGES[number];
+
+export const tasks: TelemetryTask[] = [
+  {
+    name: 'processor_events',
+    executor: async ({ indices, search }) => {
+      const indicesByProcessorEvent = {
+        error: indices['apm_oss.errorIndices'],
+        metric: indices['apm_oss.metricsIndices'],
+        span: indices['apm_oss.spanIndices'],
+        transaction: indices['apm_oss.transactionIndices'],
+        onboarding: indices['apm_oss.onboardingIndices'],
+        sourcemap: indices['apm_oss.sourcemapIndices']
+      };
+
+      type ProcessorEvent = keyof typeof indicesByProcessorEvent;
+
+      const jobs: Array<{
+        processorEvent: ProcessorEvent;
+        timeRange: TimeRange;
+      }> = flatten(
+        (Object.keys(
+          indicesByProcessorEvent
+        ) as ProcessorEvent[]).map(processorEvent =>
+          TIME_RANGES.map(timeRange => ({ processorEvent, timeRange }))
+        )
+      );
+
+      const counts = await jobs.reduce((prevJob, current) => {
+        return prevJob.then(async data => {
+          const { processorEvent, timeRange } = current;
+
+          return merge({}, data, {
+            [processorEvent]: {
+              [timeRange]: (
+                await search({
+                  index: indicesByProcessorEvent[processorEvent],
+                  body: {
+                    query: {
+                      bool: {
+                        filter: [
+                          { term: { [PROCESSOR_EVENT]: processorEvent } },
+                          ...(timeRange !== 'all'
+                            ? [
+                                {
+                                  range: {
+                                    '@timestamp': {
+                                      gte: `now-${timeRange}`
+                                    }
+                                  }
+                                }
+                              ]
+                            : [])
+                        ]
+                      }
+                    },
+                    track_total_hits: true
```

Comment:
I can't see what this and the comment below was referring too because I force-pushed 😅but I did went over the queries again and made sure I wasn't collecting any hits unless it is necessary.